cmd/kubeadm/app/cmd/phases/workflow/phase.go

```diff
@@ -71,6 +74,10 @@ type Phase struct {
 	// Nb. if two or phases have the same local flags, please consider using local flags in the parent command
 	// or additional flags defined in the phase runner.
 	LocalFlags *pflag.FlagSet
+
+	// Args defines the positional arg function to be used for validating args for this phase
+	// If not set a phase will adopt the args of the top level command.
+	Args cobra.PositionalArgs
```

Comment:
The `Args` name is awfully misleading though. The problem was started in Cobra itself. I am OK with giving it a slightly different, but better name.

Reply:
@rosti yeah, I followed Cobra, I thought anything other then what cobra did was misleading. I understand what you are saying as the first time I saw it in cobra it was confusing. What are you suggesting?

Reply:
This is up to you. If you want you can leave it like this.

Reply:
ArgsValidator is the only thing I could think of.

Reply:
+1 for ArgsValidator

Reply:
done.